lib/matplotlib/backend_tools.py

```diff
@@ -111,6 +117,23 @@ def destroy(self):
         """
         pass
 
+    def state_changed_event(self, sender, canvasevent, data):
```

Comment:
Good point, just about to make the same comment, mostly a copy paste from my old code from the summer.  I will change it, which do you think best? ``trigger_event``, ``tool_triggered_event``, or perhaps even ``fire_trigger_event``?